message_ix/model/MESSAGE/model_core.gms

```diff
@@ -577,6 +590,61 @@ $macro COMMODITY_BALANCE(node,commodity,level,year,time) (
+* for first model period (differential with historical remaining capacity)
+  + SUM( (location,tec,vintage,year_all2)$( inv_tec(tec) AND map_tec_lifetime_extended(location,tec,vintage,year_all2)                  \
+           AND NOT map_tec_lifetime_extended(location,tec,vintage,year) AND first_period(year) AND seq_period(year_all2,year)           \
+           AND (remaining_capacity_extended(location,tec,vintage,year) = 0)                                                               \
+),                                                                                                                                      \
+* output by all new capacity of technologies located at 'location' sending to 'node' and 'time' distributed over years of periods
+* Based on the condition that remaining capacity in the first year is 0. 
+        output_cap_ret(location,tec,vintage,node,commodity,level,time)                                                 \
+        *  (historical_new_capacity(node,tec,vintage) * remaining_capacity_extended(node,tec,vintage,year_all2))            \
+* input by all new capacity of technologies located at 'location' taking from 'node' and 'time' distributed over years of periods
+        - input_cap_ret(location,tec,vintage,node,commodity,level,time)                                                \
+        *  (historical_new_capacity(node,tec,vintage) * remaining_capacity_extended(node,tec,vintage,year_all2))) \      
+* commodity input and output associated with retirement of technology capacity (via differentials of capacity of successive periods)
+* for first model period (differential with historical remaining capacity)
+* for lifetimes out of duraion_period, remaining capacity is different than 1 or 0 in the first model period. 
+  + SUM( (location,tec,vintage,year_all2)$( inv_tec(tec) AND map_tec_lifetime_extended(location,tec,vintage,year_all2)                      \
+           AND first_period(year) AND seq_period(year_all2,year)                                                                             \
+           AND (remaining_capacity_extended(location,tec,vintage,year)< 1) AND (remaining_capacity_extended(location,tec,vintage,year)>0)     \
+),                                                                                                                                             \                                                                                                                           
+* output by all new capacity of technologies located at 'location' sending to 'node' and 'time' distributed over years of periods
+        output_cap_ret(location,tec,vintage,node,commodity,level,time)                                                 \
+        *  (historical_new_capacity(node,tec,vintage) * (1 - remaining_capacity_extended(node,tec,vintage,year)))            \
```

Comment:
I wasn't entirely sure why it should be `(1 - remaining_capacity_extended)` here. If `remaining_capacity_extended `= 0.3 for example, doesn't it mean the lifetime covers only 30% of the duration of the first period? Then the amount of retiring capacity is also just 30% of the `historical_new_capacity`? Pls correct me if I'm not understanding it..

Reply:
I can see Gamze's point here. So for the first period if remaining_capacity_extended=0.3, 70% is retiring during the first period. So this seems ok then. But then I wonder how the retirement of that 30% is considered in the second period. The next SUM() block below is just based on CAP differences, which cannot consider this partial retirement?